src/scheduler.rs

```diff
@@ -619,28 +605,38 @@ async fn fetch_idle(ctx: &Context, connection: &mut Imap, folder_meaning: Folder
     {
         info!(ctx, "IMAP IDLE is disabled, going to fake idle.");
         connection
-            .fake_idle(ctx, Some(watch_folder), folder_meaning)
+            .fake_idle(ctx, Some(watch_folder.clone()), folder_meaning)
             .await;
-        return;
+    } else {
+        info!(ctx, "IMAP session supports IDLE, using it.");
+        match session
+            .idle(
+                ctx,
+                connection.idle_interrupt_receiver.clone(),
+                &watch_folder,
+            )
+            .await
+            .context("idle")
+        {
+            Ok(session) => {
+                connection.session = Some(session);
+            }
+            Err(err) => {
+                connection.trigger_reconnect(ctx);
+                warn!(ctx, "{:#}", err);
+            }
+        }
     }
 
-    info!(ctx, "IMAP session supports IDLE, using it.");
-    match session
-        .idle(
-            ctx,
-            connection.idle_interrupt_receiver.clone(),
-            &watch_folder,
-        )
+    // Fetch the watched folder
+    // immediately after IDLE to reduce message delivery latency.
+    if let Err(err) = connection
+        .fetch_move_delete(ctx, &watch_folder, folder_meaning)
         .await
-        .context("idle")
+        .context("fetch_move_delete")
     {
-        Ok(session) => {
-            connection.session = Some(session);
-        }
-        Err(err) => {
-            connection.trigger_reconnect(ctx);
-            warn!(ctx, "{:#}", err);
-        }
+        connection.trigger_reconnect(ctx);
+        warn!(ctx, "{:#}", err);
     }
 }
 
```

Comment:
A few lines above it says:

```
                // Fetch the watched folder again in case scanning other folder moved messages
                // there.
                //
                // In most cases this will select the watched folder and return because there are
                // no new messages. We want to select the watched folder anyway before going IDLE
                // there, so this does not take additional protocol round-trip.
                [...] fetch_move_delete(...)
```

This can probably be removed if we fetch here.